File: Challenges/Problem_0010/Summation_of_primes.py
# KuntaNgcobo
# Summation of primes

import math

def isPrime(n):
  if (n % 2 == 0 and n != 2):
    return False
  
  factorCap = math.ceil(math.sqrt(n))+1
  for i in range(3, factorCap, 2): 
    if (n % i == 0):
      return False
  
  return True

def primeSummation(n):
  if(n < 3):
  	return 0

  total = 2
  for i in range(3, n, 2):
    if(isPrime(i)):
      total += i 
 
  return total

print(primeSummation(12))
print(primeSummation(3))
print(primeSummation(2))
print(primeSummation(1))

File: Challenges/Problem_10/Summation_of_primes.js
// KuntaNgcobo
// Summation of primes

const isPrime = n => {
  if (n % 2 === 0 && n ==! 2) {
    return false;
  }

  let factorCap = Math.ceil(Math.sqrt(n))+1;
  for (var i = 3; i < factorCap; i += 2) {
    if (n % i === 0) {
      return false;
    }
  }
  return true;
}

const primeSummation = n => {
  if(n < 3) return 0;
  let total = 2;

  for(var i = 3; i < n; i += 2){
    if(isPrime(i)){
      total += i; 
    }
  }
  return total;
}
